src/app.jsx

```diff
@@ -21,24 +21,64 @@ import cockpit from "cockpit";
 import React from "react";
 import "./app.scss";
 
-// import { Accordion, AccordionItem, AccordionContent, AccordionToggle } from "@patternfly/react-core";
+import EmptyState from "./emptyState.jsx";
+import * as service from "../lib/service.js";
 
 const _ = cockpit.gettext;
+const CERTMONGER_SERVICE_NAME = "certmonger.service";
 
 export class Application extends React.Component {
     constructor() {
         super();
-        this.state = { hostname: _("Unknown") };
+        this.state = {
+            certmongerService: undefined,
+        };
 
-        cockpit.file("/etc/hostname").watch(content => {
-            this.setState({ hostname: content.trim() });
-        });
+        this.updateCertmongerService();
```

Comment:
Anything that calls `setState` belongs in `componentDidMount`, in the constructor you can only initialize this.state.

Reply:
Fixed